repos/system_upgrade/common/libraries/distro.py

```diff
@@ -16,3 +19,199 @@ def get_distribution_data(distribution):
+            },
+        },
+        'almalinux': {
+            '8': {
+                # TODO is this true on all archs?
+                '/etc/yum.repos.d/almalinux-ha.repo': 'all',
+                '/etc/yum.repos.d/almalinux-nfv.repo': 'all',
+                '/etc/yum.repos.d/almalinux-plus.repo': 'all',
+                '/etc/yum.repos.d/almalinux-powertools.repo': 'all',
+                '/etc/yum.repos.d/almalinux-resilientstorage.repo': 'all',
+                '/etc/yum.repos.d/almalinux-rt.repo': 'all',
+                '/etc/yum.repos.d/almalinux-sap.repo': 'all',
+                '/etc/yum.repos.d/almalinux-saphana.repo': 'all',
+                '/etc/yum.repos.d/almalinux.repo': 'all',
+            },
+            '9': {
+                '/etc/yum.repos.d/almalinux-appstream.repo': 'all',
+                '/etc/yum.repos.d/almalinux-baseos.repo': 'all',
+                '/etc/yum.repos.d/almalinux-crb.repo': 'all',
+                '/etc/yum.repos.d/almalinux-extras.repo': 'all',
+                '/etc/yum.repos.d/almalinux-highavailability.repo': 'all',
+                '/etc/yum.repos.d/almalinux-plus.repo': 'all',
+                '/etc/yum.repos.d/almalinux-resilientstorage.repo': 'all',
+                '/etc/yum.repos.d/almalinux-sap.repo': 'all',
+                '/etc/yum.repos.d/almalinux-saphana.repo': 'all',
+                # RT and NFV are only on x86_64 on almalinux 9
+                '/etc/yum.repos.d/almalinux-nfv.repo': ['x86_64'],
+                '/etc/yum.repos.d/almalinux-rt.repo': ['x86_64'],
+            },
+            '10': {
+                # no resilientstorage on 10
+                '/etc/yum.repos.d/almalinux-appstream.repo': 'all',
+                '/etc/yum.repos.d/almalinux-baseos.repo': 'all',
+                '/etc/yum.repos.d/almalinux-crb.repo': 'all',
+                '/etc/yum.repos.d/almalinux-extras.repo': 'all',
+                '/etc/yum.repos.d/almalinux-highavailability.repo': 'all',
+                '/etc/yum.repos.d/almalinux-plus.repo': 'all',
+                '/etc/yum.repos.d/almalinux-sap.repo': 'all',
+                '/etc/yum.repos.d/almalinux-saphana.repo': 'all',
+                # RT and NFV are only on x86_64 on almalinux 10
+                '/etc/yum.repos.d/almalinux-nfv.repo': ['x86_64'],
+                '/etc/yum.repos.d/almalinux-rt.repo': ['x86_64'],
+            },
+        },
+    }
+
+    repofiles = []
+    distro_repofiles = distro_repofiles_map.get(distro)
+    if not distro_repofiles:
+        raise UnknownDistributionError
+
+    # if it's not mapped it's probably an error
+    # on our side, so let's warn and return empty
+    version_repofiles = distro_repofiles.get(major_version, {})
+    api.current_logger().warning(
+        'No distro repofiles mapped for {} for major version {}'.format(
+            distro, major_version
+        )
+    )
+
+    for repofile, archs in enumerate(version_repofiles):
+        if archs is str and archs is all:
```

Comment:
I tested the code on C9 VM. I seems like this line will not work, failing with ValueError as those conditions are not met. `archs is str` will always be False, because `is` checks for object identity, not type. Also `archs is all` is not checking string content, should be `if archs == 'all'`.